### PR TITLE
BREAKING CHANGE: provide possibility to run subtests and skip steps

### PR DESCRIPTION
**Pull request by stambata**
ut-run sequence now returns promise

### CODE REVIEW

**Comment by kalinkrustev** [comment]
I was not able to find out what exactly is the breaking change. 
Introducing bugs (if any) is not a breaking change, as they will be fixed in next patch version.

A suggestion for better output formatting of the nesting: if possible, the following can be better:
test 1
test 2\subtest 1
test 2\subtest 2

Optionally we can consider removing when from test.js, if we anyway fear that some unstable versions of the module will be published.

**Comment by stambata** [comment]
Good point. Changing the printing to: step\subtest and removing 'when' are definitely things I'll do along with the bug fix before proceeding with the merge.
And about the breaking change - the only thing that will break the old tests is the requirement to always return the promise that the sequence returns (otherwise the test will not finish).
I.e. When writing tests:
```
steps: function (test, bus, run) {
   run(test, bus, flow);
}
```
should become:
```
steps: function (test, bus, run) {
   return run(test, bus, flow);
}
```

It is just one return that should be added per test.... but it is still a change that need to be applied in order for the tests to pass correctly.

**Comment by kalinkrustev** [comment]
> to always return the promise
So in that case can we throw big error with stack, so that we catch these places where it was not returned?

**Comment by stambata** [comment]
* Dependency to when has been removed from ut-run/test
* Prining has been improved
* Bugs with error absorbing and tests interruption have been resolved

@venetahristova Please check if the changes you requested have been correctly implemented

File: test.js [hunk 1]
```diff
@@ -1,90 +1,120 @@
 // var log = require('why-is-node-running');
 var tape = require('blue-tape');
 var run = require('./index').runParams;
-var when = require('when');
 var loadtest = require('loadtest');
 
+function lift(fn) {
```
Comment (kalinkrustev) [inline test.js]: why not simply:
```javascript
const promisify = fn =>function(){
    return new Promise(resolve => resolve(fn.apply(this, arguments)));
};
```
Note that your fn.apply changes `this` to `fn`, which I believe is an error.

Also maybe change the name from `lift`, as in original code `when.lift` was an error, as `when.lift` supposed to be used on node callback style functions, while ours were not such.

Comment (stambata) [inline test.js]: Please see latest commit: https://github.com/softwaregroup-bg/ut-run/pull/11/commits/6397abf6e0e2af7fcff1571e34d9d24e869b1394

**Pull request merged**